extensions/objects/templates/set-of-algebraic-identifier-editor.component.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for set of algebraic identifier editor.
+ */
+
+// Every editor directive should implement an alwaysEditable option. There
+// may be additional customization options for the editor that should be passed
+// in via initArgs.
+
+require('services/guppy-initialization.service.ts');
+
+angular.module('oppia').component('setOfAlgebraicIdentifierEditor', {
+  bindings: {
+    value: '='
+  },
+  template: require('./list-editor.directive.html'),
+  controller: ['GuppyInitializationService',
+    function(GuppyInitializationService) {
+      const ctrl = this;
+
+      ctrl.$onInit = function() {
+        ctrl.SCHEMA = {
+          type: 'list',
+          items: {
+            type: 'unicode',
+            choices: GuppyInitializationService.customOskLetters
```

Comment:
Ok, so this editor initializes with the OSK letters, but then the creator can edit that?

Reply:
Yes, if and when the creator edits the customization arg, these choices in the dropdown will be accordingly updated.